src/pip/_internal/commands/list.py

```diff
@@ -5,9 +5,11 @@
 
 import json
 import logging
+import concurrent.futures
```

Comment:
This isn't Python 2.7 compatible. I'm not against having this improvement only apply for Python 3, but we do need to gracefully fall back to the existing behaviour if we do that. Alternatively we could have a Python 2.7 compatible implementation using raw threading, but I'm fine if you consider that too much work for an old version of Python. See [here](https://pip.pypa.io/en/stable/development/release-process/#python-2-support) for pip's Python 2 support policy.

Reply:
Thanks for your fast reply. I understood that this is not Python 2.7 compatible but thanks for pointing it. I wanted to check tests at first on Python 3. In my opinion we can just add `concurrent.futures` as vendor library. Because I don't see any Python 2 incompatible modules in [Python 3 implementation](https://github.com/python/cpython/blob/3.8/Lib/concurrent/futures/thread.py) of `ThreadPoolExecutor`. I will try this solution.

Reply:
Agreed, that looks like a reaonable approach.

Reply:
There is `multiprocessing.dummy` which is compatible with Python 2.

Reply:
@McSinyx Thank you.
@pfmoore I am ready to merge.